kivy/core/window/window_sdl2.py

```diff
@@ -489,6 +489,12 @@ def _get_window_pos(self):
     def _set_window_pos(self, x, y):
         self._win.set_window_pos(x, y)
 
+    def _get_window_opacity(self):
+        return self._win.get_window_opacity()
+
+    def _set_window_opacity(self, opacity):
+        self._win.set_window_opacity(opacity)
```

Comment:
Change to:
```python
if self.opacity != opacity:
    return self._win.set_window_opacity(opacity)
```

This way `self.opacity` will use the cached value and if `True` is returned by `set_window_opacity` opacity getter method will be called to get new value which be cached.

Reply:
Done.